slatedb/src/manifest/store.rs

```diff
@@ -248,19 +205,14 @@ impl FenceableManifest {
 // can use the `refresh` method to refresh the locally stored manifest+id with the latest
 // manifest stored in the object store.
 pub(crate) struct StoredManifest {
-    id: u64,
-    manifest: Manifest,
-    manifest_store: Arc<ManifestStore>,
+    inner: StoredRecord<Manifest>,
 }
 
 impl StoredManifest {
     async fn init(store: Arc<ManifestStore>, manifest: Manifest) -> Result<Self, SlateDBError> {
-        store.write_manifest(1, &manifest).await?;
-        Ok(Self {
-            id: 1,
-            manifest,
-            manifest_store: store,
-        })
+        // Preserve original behavior: write via ManifestStore (object-store path and semantics)
+        let inner = StoredRecord::init(Arc::clone(&store.inner), manifest.clone()).await?;
```

Comment:
This doesn't feel right. ManifestStore holds:

```
inner: Arc<RecordStore<Manifest>>,
```

Can't you pass that in instead of `store: Arc<ManifestStore>`?